### PR TITLE
Refactor/simplify wikigeo extraction

### PR DESCRIPTION
**Pull request by jasonbaldridge**
Just a bit of code clean up while regenerating Wikigeo files for myself!

- Got rid of the OSM variant of the executable for extracting data. Now, adding --osm_path to extract_wikigeo_contexts.py does OSM extraction, otherwise it is ignored.
- Instead of parsing a path like 'dc.json', caller needs to give --output_dir, then the filenames are lowercase on the region plus split name plus '.json'. E.g. if the region is DC, the output is dc_train.json, etc.
- split_datasets() returns a dictionary with split names as keys. This makes it clear what is what and then makes it easier to loop over the splits and output them in extract_wikigeo_contexts.py.
- Added types, e.g. Sequence[wikigeo.WikigeoEntity] rather than bare Sequence.
- Changed Text to str (as per recommendations for Py3 code).

### CODE REVIEW

File: cabby/data/extract_wikigeo_contexts.py [hunk 1]
```diff
@@ -11,12 +11,16 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+'''Command line method to extract Wikipedia and Wikidata items and save to file.
```
Comment (tzufgoogle) [inline cabby/data/extract_wikigeo_contexts.py:14]: And OSM:
Command line method to extract Wikipedia, Wikidata, and OSM items, and construct WikiGeo samples and save to file.

File: cabby/data/extract_wikigeo_contexts.py [hunk 2]
```diff
@@ -28,53 +32,46 @@
...
 from cabby.geo import regions
 
-
-FLAGS = flags.FLAGS
 flags.DEFINE_enum(
   "region", None, regions.ALLOWED_REGIONS,
-  "Map areas: Manhattan, Pittsburgh or Pittsburgh_small.")
-flags.DEFINE_string("path", None, "The path where the data will be saved.")
-
+  "Map areas: Manhattan, Pittsburgh or DC.")
```
Comment (tzufgoogle) [inline cabby/data/extract_wikigeo_contexts.py:37]: I removed DC query as the current DC region doesn't return Wikidata items. 
Instead - Pittsburgh_small

**Review by tzufgoogle — COMMENTED** [review]
Two small remarks on the comments.

File: cabby/data/extract_wikigeo_contexts.py (hunk 1 above)
Comment (jasonbaldridge) [reply cabby/data/extract_wikigeo_contexts.py]: Fixed, thanks!

File: cabby/data/extract_wikigeo_contexts.py (hunk 2 above)
Comment (jasonbaldridge) [reply cabby/data/extract_wikigeo_contexts.py:37]: Changed example to Pittsburgh_small

Comment (tzufgoogle) [reply cabby/data/extract_wikigeo_contexts.py:37]: I still see - "Map areas: Manhattan, Pittsburgh or DC."

**Review by tzufgoogle — COMMENTED** [review]
One small thing left

**Review by tzufgoogle — APPROVED** [review]
LGTM

**Pull request merged**